source/features/conflict-marker.tsx

```diff
@@ -0,0 +1,90 @@
+import select from 'select-dom';
+import React from 'dom-chef';
+import * as api from '../libs/api';
+import {getOwnerAndRepo, getRepoURL} from '../libs/utils';
+import features from '../libs/features';
+import {alert} from '../libs/icons';
+
+interface PRConfig {
+	id: string;
+	owner: string;
+	name: string;
+}
+
+function getPrNumber(pr: string): string {
+	return pr.split('_')[1];
+}
+
+function createQueryFragment(pr: PRConfig): string {
+	return `
+		${api.escapeKey(pr.id)}: repository(owner: "${pr.owner}", name: "${pr.name}") {
+			pullRequest(number: ${getPrNumber(pr.id)}) {
+				mergeable
+			}
+		}
+	`;
+}
+
+function buildQuery(
+	prs: PRConfig[]
+): string {
+	return prs.map(createQueryFragment).join('\n');
+}
+
+function getPRConfig(element: HTMLElement): PRConfig {
+	try {
+		const prTitle = select('a[data-hovercard-type="repository"]', element)!.textContent!;
+		const [owner, name] = prTitle.trim().split('/');
+
+		return {
+			id: element.id,
+			owner,
+			name
+		};
+	} catch (error) {
+		const {ownerName, repoName} = getOwnerAndRepo();
+
+		return {
+			id: element.id,
+			owner: ownerName,
+			name: repoName
+		};
+	}
+}
```

Comment:
Each PR has a link to the PR and it has all of this information. You can pass the `.pathname` to `createQueryFragment` and get the info in there:

```js
const [, user, repo,,number] = pathname.split('/');
```

Therefore you can merge `getPRConfig`, `buildQuery` and `createQueryFragment`

Reply:
Those pieces of information are unavailable on the `global issues` page. So to make it work on `https://github.com/issues` I have to get this information from DOM nodes.

Reply:
They are. Each link to the issues and PRs has that information. In my examples, `pathname` refers to each link's `.pathname` attribute, not `location.pathname`